src/main/java/CmdHandler.java

```diff
@@ -45,20 +56,26 @@ private void handleUnMark(int i) {
     void handle() {
         Scanner sc = new Scanner(System.in);
         while (!done) {
-            String inputStr = sc.nextLine();
-            String[] inputArr = inputStr.split(" ");
-            if (inputArr[0].equals("list")) {
-                handleList();
-            } else if (inputArr[0].equals("bye")) {
-                handleBye();
-            } else if (inputArr[0].equals("unmark")) {
-                handleUnMark(Integer.parseInt(inputArr[inputArr.length - 1]) - 1);
-            } else if (inputArr[0].equals("mark")) {
-                handleMark(Integer.parseInt(inputArr[inputArr.length - 1]) - 1);
-            } else {
-                addTask(inputStr);
+            try {
+                String inputStr = sc.nextLine();
+                String[] inputArr = inputStr.split(" ");
+                if (inputArr[0].equals("list")) {
+                    handleList();
+                } else if (inputArr[0].equals("bye")) {
+                    handleBye();
+                } else if (inputArr[0].equals("unmark")) {
+                    handleUnMark(Integer.parseInt(inputArr[inputArr.length - 1]) - 1);
+                } else if (inputArr[0].equals("mark")) {
+                    handleMark(Integer.parseInt(inputArr[inputArr.length - 1]) - 1);
+                } else {
+                    addTask(inputStr);
+                }
+                System.out.println();
+
+            } catch (DukeException de) {
+                System.out.println(de.getMessage());
+
```

Comment:
The try and catch implementation of the exception handler and you consider about another exception class just to handle the command error. :+1: